engine/events/overworld.asm

```diff
@@ -0,0 +1,100 @@
+INCLUDE "constants.asm"
+
+SECTION "TeleportFunction", ROMX[$52db], BANK[$03]
+
+TeleportFunction: ; 03:52db
+	xor a
+	ld [wFieldMoveScriptID], a
+.loop
+	ld a, [wFieldMoveScriptID]
+	bit 7, a
+	jr nz, .finish
+	ld hl, .JumpTable
+	call CallJumptable
+	jr .loop
+.finish
+	and $7f
+	ld [wFieldMoveSucceeded], a
+	ret
+	
+.JumpTable
+	dw .TryTeleport
+	dw .DoTeleport
+	dw .FailTeleport
+	dw .CheckIfSpawnPoint
+
+.TryTeleport: ; 03:52fc
+	call GetMapEnvironment
+	cp TOWN
+	jr z, .success
+	cp ROUTE
+	jr z, .success
+	ld a, $02

```

Comment:
Create constants for these, even if they're just MOVE_SCRIPT_02 or such. (Please check if pokecrystal has something similar)